<?xml version="1.0" ?>
<robot name="7th_axe_robot" xmlns:xacro="http://wiki.ros.org/xacro">
  <xacro:include filename="$(find fanuc_resources)/urdf/common_materials.xacro"/>


  <!-- links: main serial chain  de fanuc-->
    <link name="base_link">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/visual/base_link.stl"/>
        </geometry>
        <xacro:material_fanuc_gray24 />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/collision/base_link.stl"/>
        </geometry>
      </collision>
    </link>
    <link name="link_1">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/visual/link_1.stl"/>
        </geometry>
        <xacro:material_fanuc_white />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/collision/link_1.stl"/>
        </geometry>
      </collision>
    </link>
    <link name="link_2">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/visual/link_2.stl"/>
        </geometry>
        <xacro:material_fanuc_white />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/collision/link_2.stl"/>
        </geometry>
      </collision>
    </link>
    <link name="link_3">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/visual/link_3.stl"/>
        </geometry>
        <xacro:material_fanuc_white />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/collision/link_3.stl"/>
        </geometry>
      </collision>
    </link>
    <link name="link_4">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/visual/link_4.stl"/>
        </geometry>
        <xacro:material_fanuc_white />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/collision/link_4.stl"/>
        </geometry>
      </collision>
    </link>
    <link name="link_5">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/visual/link_5.stl"/>
        </geometry>
        <xacro:material_fanuc_white />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/collision/link_5.stl"/>
        </geometry>
      </collision>
    </link>
    <!-- Link_6 original du Fanuc-->
    <!--link name="link_6">
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/visual/link_6.stl"/>
        </geometry>
        <xacro:material_fanuc_gray24 />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="package://fanuc_crx10ia_support/meshes/crx10ial/collision/link_6.stl"/>
        </geometry>
      </collision>
    </link-->
    <!-- joints: main serial chain -->
    <joint name="joint_1" type="revolute">
      <origin xyz="0 0 0.245" rpy="0 0 0" />
      <parent link="base_link" />
      <child link="link_1" />
      <axis xyz="0 0 1" />
      <limit effort="0" lower="${radians(-180)}" upper="${radians(180)}" velocity="${radians(120)}" />
    </joint>
    <joint name="joint_2" type="revolute">
      <origin xyz="0 0 0" rpy="0 0 0" />
      <parent link="link_1" />
      <child link="link_2" />
      <axis xyz="0 1 0" />
      <limit effort="0" lower="${radians(-180)}" upper="${radians(180)}" velocity="${radians(120)}" />
    </joint>
    <joint name="joint_3" type="revolute">
      <origin xyz="0 0 0.710" rpy="0 0 0" />
      <parent link="link_2" />
      <child link="link_3" />
      <axis xyz="0 -1 0" />
      <limit effort="0" lower="${radians(-270)}" upper="${radians(270)}" velocity="${radians(180)}" />
    </joint>
    <joint name="joint_4" type="revolute">
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <parent link="link_3"/>
      <child link="link_4"/>
      <axis xyz="-1 0 0"/>
      <limit effort="0" lower="${radians(-190)}" upper="${radians(190)}" velocity="${radians(180)}" />
    </joint>
    <joint name="joint_5" type="revolute">
      <origin xyz="0.540 -0.150 0" rpy="0 0 0"/>
      <parent link="link_4"/>
      <child link="link_5"/>
      <axis xyz="0 -1 0"/>
      <limit effort="0" lower="${radians(-180)}" upper="${radians(180)}" velocity="${radians(180)}" />
    </joint>
    <!-- Joint 6 original du robot-->
    <!--joint name="joint_6" type="revolute">
      <origin xyz="0.160 0 0" rpy="0 0 0"/>
      <parent link="link_5"/>
      <child link="link_6"/>
      <axis xyz="-1 0 0"/>
      <limit effort="0" lower="${radians(-190)}" upper="${radians(190)}" velocity="${radians(180)}" />
    </joint-->


  <!-- Ajout du 7th joint -->
  <link name="husky"/>
  
  <joint name="husky_to_robot_base_plate" type="prismatic">
    <origin xyz="0.0 0.0 0.798396" rpy="0 0 0"/>
    <parent link="husky"/>
    <child  link="base_link"/>
    <axis xyz="0 0 1"/>
    <limit effort="4000.0" velocity="0.150" lower="0.0" upper="1.050"/>
    <!--TODO safety controller/-->
  </joint>

  <!--Define a new joint_6 TCP with festool  in one joint -->
  <link name="TCP"/>

  <joint name="joint_6" type="revolute">
      <origin xyz="0.160 -0.020860 0.141701" rpy="${pi} ${-pi/2.0} 0"/>
      <parent link="link_5"/>
      <child link="TCP"/>
      <axis xyz="-1 0 0"/>
      <limit effort="0" lower="${radians(-190)}" upper="${radians(190)}" velocity="${radians(180)}" />
    </joint>

</robot>